lib/remotus/winrm_connection.rb

```diff
@@ -69,14 +70,16 @@ def base_connection(reload: false)
 
     #
     # Retrieves/creates the WinRM shell connection for the host
+    #
+    # @param [symbol] shell connection shell type, defaults to :powershell
     # If the connection already exists, the existing connection will be retrieved
     #
-    # @return [WinRM::Shells::Powershell] remote connection
+    # @return [WinRM::Shells::Powershell, WinRM::Shells::Elevated] remote connection
     #
-    def connection
+    def connection(shell = :powershell)
       return @connection unless restart_connection?
```

Comment:
The `restart_connection?` method will need to be updated to return true in cases where the shell has been changed since the connection was last cached.